Mathlib/Analysis/Complex/Hadamard.lean

```diff
@@ -0,0 +1,487 @@
+
+-/
+
+
+open Set Filter Function Complex
+
+namespace Complex
+namespace HadamardThreeLines
+variable (z : ℂ)
+
+/-- The strip in the complex plane containing all `z ∈ ℂ` such that `z.re ∈ Ioo a b`. -/
+def strip (a : ℝ) (b : ℝ) : Set ℂ := re ⁻¹' Ioo a b
+
+/-- The strip in the complex plane containing all `z ∈ ℂ` such that `z.re ∈ Icc a b`. -/
+def closedStrip (a : ℝ) (b : ℝ) : Set ℂ := re ⁻¹' Icc a b
+
+/-- The strip in the complex plane containing all `z ∈ ℂ` such that `z.re ∈ Ioo a b` and
+`z.im ∈ Ioo (-T) T`. -/
+def bddStrip (a : ℝ) (b : ℝ) (T : ℝ) : Set ℂ :=  re ⁻¹' Ioo a b ∩ im ⁻¹' Ioo (-T) T
+
+/-- The strip in the complex plane containing all `z ∈ ℂ` such that `z.re ∈ Icc a b` and
+`z.im ∈ Icc (-T) T`. -/
+def closedBddStrip (a : ℝ) (b : ℝ) (T : ℝ) : Set ℂ :=  re ⁻¹' Icc a b ∩ im ⁻¹' Icc (-T) T
+
+/-- The filter along the vertical strip `re ⁻¹' Ioo a b` and `|z.im| atTop` -/
+def cocompactStrip (a b : ℝ) : Filter ℂ :=
+  comap im (cocompact ℝ) ⊓ (comap re (principal (Icc a b)))
+
+/-- The supremum of the absolute value of `f` on imaginary lines. (Fixed real part) -/
+noncomputable def sSupAbsIm (f : ℂ → ℂ) (x : ℝ) : ℝ  :=  sSup ((abs ∘ f) '' (re ⁻¹' {x}))
+
+
+/-- The inverse of the interpolation of `sSupAbsIm` on the two boundaries. -/
+noncomputable def invInterpStrip (f : ℂ → ℂ) (z : ℂ) : ℂ :=
+  if (sSupAbsIm f (0 : ℝ)) = (0 : ℝ) ∨ (sSupAbsIm f (1 : ℝ)) = (0 : ℝ)
+    then 0
+    else (sSupAbsIm f 0)^(z-1) * (sSupAbsIm f 1)^(-z)
+
+/-- A function useful for the proofs steps. We will aim to show that it is bounded by 1. -/
+private noncomputable def F (f : ℂ → ℂ) := fun z ↦ f z • invInterpStrip f z
+
+/-- Similar to `F` only 'easier'. Useful for proof steps. -/
+private noncomputable def F' (n : ℕ) (f : ℂ → ℂ) := fun z ↦ F f z • exp ((z^2-1) * (n : ℝ)⁻¹)
+
+
+-- Small lemma : Sup of abs is nonneg
+lemma sSupAbsIm_nonneg (f : ℂ → ℂ) (x : ℝ) : 0 ≤ sSupAbsIm f x := by
+  apply Real.sSup_nonneg
+  rintro y ⟨z1, _, hz2⟩
+  simp only [← hz2, comp, map_nonneg]
+
+
+-- Definition rewrites for invInterpStrip
+lemma invInterpStrip_eq_of_pos (f : ℂ → ℂ ) (h0 : 0 < sSupAbsIm f 0) (h1 : 0 < sSupAbsIm f 1) :
+    invInterpStrip f z = (sSupAbsIm f 0)^(z-1) * (sSupAbsIm f 1)^(-z) := by
+  simp only [ne_of_gt h0, ne_of_gt h1, invInterpStrip, if_false, or_false]
+
+
+lemma invInterpStrip_eq_of_zero (f : ℂ → ℂ ) (h : (sSupAbsIm f 0) = 0 ∨ (sSupAbsIm f 1) = 0) :
+    invInterpStrip f z = 0 :=
+  if_pos h
+
+-- Differentiable continuous function invInterpStrip
+lemma diffContOnCl_invInterpStrip (f : ℂ → ℂ) : DiffContOnCl ℂ (invInterpStrip f) (strip 0 1) := by
+  by_cases (sSupAbsIm f 0) = 0 ∨ (sSupAbsIm f 1) = 0
+  -- Case everywhere 0
+  -- Starting with Lemma to handle rewriting of invInterpStrip as a function.
+  · have hzero : invInterpStrip f = 0 := by
+      rw [funext_iff]
+      intro z
+      simp only [invInterpStrip_eq_of_zero z f h, Pi.zero_apply, eq_self_iff_true]
+    rw [hzero, Pi.zero_def]
+    exact diffContOnCl_const
+
```

Comment:
no blank lines inside proofs. The comments are already enough to separate the parts of the proof (and it's very good that you put these comments, they help a lot follow the logic!)